server/middleware/user.js

```diff
@@ -35,6 +35,40 @@ class User {
     }
   }
 
+  /**
+   * This method verifies whether email is registered.
+   * @param {object} req the user's request.
+   * @param {object} res the response.
+   * @param {Function} next pass to next function.
+   * @returns {object} message indicating used email.
+   */
+  static async verifyRegisteredEmail(req, res, next) {
+    try {
+      const { email } = req.body;
+      const registeredEmail = await UserHelper.findUser({ email });
+      if (registeredEmail === null) {
+        res.status(404).send({
+          status: 404,
+          message: 'Email is not registered in our system.'
+        });
+      } else if (registeredEmail.dataValues.isVerified === true) {
+        next();
+      } else {
+        res.status(400).send({
+          status: 400,
+          message: 'You should verify your email before you reset it.'
```

Comment:
Instead of **it**, put **the password**. Thus, the message will become **You should verify your email before you reset the password**.